management-app/src/main/java/com/nearsoft/fgaribay/mgmt/ManagementController.java

```diff
@@ -0,0 +1,40 @@
+package com.nearsoft.fgaribay.mgmt;
+
+import com.nearsoft.fgaribay.mgmt.model.Product;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+import javax.validation.Valid;
+import java.util.ArrayList;
+import java.util.List;
+
+@Controller
+@RequestMapping(value = "/products")
+public class ManagementController {
+    private static List<Product> products = new ArrayList<>();
+
+    @Autowired
+    private BrokerConnector brokerConnector;
+
+    @RequestMapping(value = {"/", "/list"}, method = RequestMethod.GET)
+    public String viewPersonList(Model model) {
+        model.addAttribute("products", brokerConnector.retrieveProductList());
+        return "listProducts";
+    }
+
+    @RequestMapping(value = "/create", method = RequestMethod.GET)
+    public String greetingForm(Model model) {
+        model.addAttribute("product", new Product());
+        return "createProduct";
+    }
+
+    @RequestMapping(value = "/create", method = RequestMethod.POST)
+    public String submit(@Valid @ModelAttribute("product") Product product) {
+        brokerConnector.createProduct(product);
```

Comment:
remember to validate the product, you should probably move the brokerConnector to a service layer, where you would make a validation before sending it to the queue, otherwise if the validation is done in the microservice, how are you gonna return the error to the mgmt app?

Reply:
Thanks, you're right, the app should make proper validations. Do you think this is what I should be looking into for implementing this? (https://docs.spring.io/spring/docs/4.1.x/spring-framework-reference/html/validation.html)

I'm thinking of adding a "price" field to the product and have that require some validation logic. As for the strings, I plan on checking for nulls, lengths and only allow alphanumerics and punctuation marks (instead of escaping, to keep the data layer in a black box). Any other suggestions?

Would you prefer to see these changes in a new PR or on this same one?

Reply:
You can use that, there's also a Hibernate validator which uses annotations, but you could also just validate with simple IFs directly checking the object, it's not an issue for me, leave a more complex validation if you want for last, if you have time. And yes, you can put it in a new PR :)